src/transformers/models/beit/modeling_beit.py

```diff
@@ -196,12 +197,16 @@ def forward(
         self,
         pixel_values: torch.Tensor,
         bool_masked_pos: Optional[torch.BoolTensor] = None,
-        interpolate_pos_encoding: bool = False,
+        interpolate_pos_encoding: Optional[bool] = None,
     ) -> torch.Tensor:
+        if self.position_embeddings is not None and interpolate_pos_encoding is not None:
+            warnings.warn(
```

Comment:
I kind remember, for logic inside `forward`, we would rather using `logger.warning_once()` ... to avoid a lot of spam log message (because `forward` is usually called many times).

WDYT?

Reply:
hmm, no problem switching to logger, but `warning.warn` also warn just once for the same message, a simple reproducer:

```python
import warnings

for i in range(10):
    warnings.warn("test")
```

output
```
/home/ubuntu/projects/transformers/test_warnings.py:4: UserWarning: test
  warnings.warn("test")
```

Reply:
oh, ok!